### PR TITLE
util changes and mail template func

### PR DESCRIPTION
**Pull request by pythexcel**

### CODE REVIEW

File: app/util.py [hunk 1]
```diff
@@ -48,79 +49,134 @@ def validate_message(user=None,message=None,req_json=None,message_detail=None):
...
-        if 'user' in slack_user_detail:
-            slack = slack_id(slack_user_detail['user']['email'])
-            slack_user_detail['user'] = "<@" + slack + ">!"
-        else:
-            pass            
-        message_str = message
-        for data in message_variables:
-            if data in slack_user_detail:
-                message_str = message_str.replace("@"+data+":", slack_user_detail[data])
+        print("sdasd")
```
Comment (reactexcel) [inline app/util.py]: remove these prints . don't send me pr with such stuff

File: app/api/message_create.py [hunk 2]
```diff
@@ -37,7 +40,56 @@ def notification_message(message_origin):
...
+                "channels":channel,
+                "for_email": for_email,
+                "for_slack": for_slack,
+                "for_phone": for_phone
+            }
+        },upsert=True)
+        return jsonify(str(ret))
+
+# This api was maded bcoz HR have some predefined variables which now i am storing in this system with their value and name
+@bp.route('/special_variable', methods=["GET","PUT"])
```
[the remark this answers is not in the document]
Comment (pythexcel) [reply app/api/message_create.py]: sir like ceo signature or
page header value
page footer value 
break line value

File: app/util.py (hunk 1 above)
Comment (pythexcel) [reply app/util.py]: ok sir

File: app/util.py [hunk 3]
```diff
@@ -48,79 +49,134 @@ def validate_message(user=None,message=None,req_json=None,message_detail=None):
...
+            else:
+                pass
         else:
             pass
     else:
         pass
 
+
+# this function will send back variables of html templates with variable from templates if there are None in special variables collection
+def special(user):
```
[the remark this answers is not in the document]
Comment (pythexcel) [reply app/util.py]: ok sir will do

File: app/api/message_create.py [hunk 4]
```diff
@@ -37,7 +40,56 @@ def notification_message(message_origin):
...
+                "variable_type":variable_type
+            }
+        },upsert=True)
+        return jsonify(str(ret))        
+
+# This will return  and add all the mail templates and which are required from HR 
+@bp.route('/get_email_template', methods=["GET", "PUT"])
+def mail_message():
+    if request.method == "GET":
+        ret = mongo.db.mail_template.find({})
```
[the remark this answers is not in the document]
Comment (pythexcel) [reply app/api/message_create.py]: ok sir will add query for all three 3 systems

**Pull request merged**